modules/node_modules/webpack_entry/Root.js

```diff
@@ -47,9 +47,15 @@ if (!Relay.Store._storeData._networkLayer._implementation) {
       }),
       // Add hash id to request
       next => req => {
-        const hash = md5([req.relayReqObj._printedQuery.text, JSON.stringify(req.relayReqObj._printedQuery.variables)].join(':'));
-        req.url = `${req.url}?hash=${hash}`;
-        return next(req);
+        const [url, search = ''] = req.url.split('?');
+        const query = {
+          ...parse(search),
+          hash: md5([
+            req.relayReqObj._printedQuery.text,
+            JSON.stringify(req.relayReqObj._printedQuery.variables),
+          ].join(':')),
+        };
+        return next({ ...req, url: `${url}?${stringify(query)}` });
```

Comment:
this is good.

but for simplicity/speed it is probably fine to just add the hash and ignore the other search params ( since there shouldn't be any) and mutate the req instead of spreading.